src/GdbServer.cc

```diff
@@ -957,7 +957,7 @@ void GdbServer::maybe_notify_stop(const GdbRequest& req,
     LOG(debug) << "Stopping for signal " << stop_siginfo;
   }
   if (is_last_thread_exit(break_status)) {
-    if (break_status.task->session().is_diversion()) {
+    if (!break_status.task || break_status.task->session().is_diversion()) {
```

Comment:
As above, this strikes me as being something that may either cause bugs or bouts of intense head-scratching in the future. Again, I'm wondering if either `BreakStatus` could be extended so `DiversionSession::diversion_step` can specify that this was an exit in a diversion session or some comments added to specify `DiversionSession` special-casing.

Reply:
Yeah, same.